### ISSUE TITLE
Datepicker is not allowing clearing date

### ISSUE THREAD

**Issue by bvedad** [themesberg/flowbite-react#1366]
- [X] I have searched the [Issues](https://github.com/themesberg/flowbite-react/issues) to see if this bug has already been reported
- [X] I have tested the latest version

## Summary

Currently, the Datepicker component in Flowbite-React allows users to select a date, but lacks a fully functional clear button that can reset the date to null. The clear button only resets the date to the original value. I propose adding a flag like isClearAction to the onSelectedDateChanged callback function. This flag would be set to true when the clear button is used, allowing developers to explicitly handle the clear action by setting the date to null if desired.

Example implementation:
```
<Controller
  control={control}
  name="birthday"
  render={({ field: { onChange, value } }) => (
    <Datepicker
      onSelectedDateChanged={(date, isClear) => {
        if (isClear) {
          onChange(null); // Explicitly handle clear action
          return;
        }
        onChange(date); // Handle date change normally
      }}
      placeholder="Please enter date"
      ref={elementRef}
      value={value ? moment(value).format('YYYY-MM-DD') : ''}
    />
  )}
/>

```
This feature would enable developers to easily implement scenarios where a user might need to completely remove a previously selected date.

## Context

In various applications, especially in forms involving date entries like event planning, user registrations, or bookings, having the capability to clear a date completely is crucial. Currently, without this feature, users of our application cannot remove a date once selected; they can only change it. This limitation affects the flexibility and user experience of the application. By implementing this feature, we will provide a more robust and intuitive interaction for users who need to reset their date selections entirely.

**Comment by OrlandisCS** [comment]
That's right, it's a problem that I faced and I had to change the date selector for something so basic, I hope it can be solved since I really like this design

**Comment by MAKINWARA** [comment]
Could you please provide any updates on this?

**Comment by lyczkop** [comment]
I had same issue and for me works when I added `key` props to `Datepicker`. That forced change in UI.
```
<Datepicker
   key={value?.toISOString()}
.....
/>
```
